File: src/MixNetFx/FacFunctions/Dialogs/WorkGroupAnalysisRelated/GetFrmLabourValueInfoCenter.vb
Imports System.Windows.Forms
Imports Facesso.Data
Imports System.Data.SqlClient

Public Class GetFrmWorkGroupAnalysis
    Inherits FacessoFunctionBase

    Public Overrides ReadOnly Property RolePermission() As IRolePermissionInfo
        Get
            Return New FacessoRolePermissionInfo(ClearanceLevel.PrintReportsOnProductionData)
        End Get
    End Property

    Public Overrides ReadOnly Property RolePermissionViolationMessage() As String
        Get
            Return My.Resources.WorkGroupAnalysis_DeniedDueToRole
        End Get
    End Property

    Public Function ShowDialog() As DialogResult
        Dim locFH As GetFrmWorkGroupAnalysis = FunctionHandler(Of GetFrmWorkGroupAnalysis).GetFunctionInstance
        If locFH Is Nothing Then Return DialogResult.None

        Dim locFrmWorkGroupAnalysis As New frmWorkGroupAnalysis
        Return locFrmWorkGroupAnalysis.ShowDialog()
    End Function
End Class

File: src/NetFxCSharp/Facesso.Functions/Dialogs/WorkGroupAnalysisRelated/GetFrmLabourValueInfoCenter.cs
using Facesso.Data;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Diagnostics;
using System.Linq;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Xml.Linq;

namespace Facesso.Functions
{
    public class GetFrmWorkGroupAnalysis : FacessoFunctionBase
    {
        public override IRolePermissionInfo RolePermission
        {
            get
            {
                return new FacessoRolePermissionInfo(ClearanceLevel.PrintReportsOnProductionData);
            }
        }

        public override string RolePermissionViolationMessage
        {
            get
            {
                return Facesso.Functions.My.Resources.WorkGroupAnalysis_DeniedDueToRole;
            }
        }

        public DialogResult ShowDialog()
        {
            GetFrmWorkGroupAnalysis locFH = FunctionHandler<GetFrmWorkGroupAnalysis>.GetFunctionInstance();
            if (locFH == null)
            {
                return DialogResult.None;
            }

            frmWorkGroupAnalysis locFrmWorkGroupAnalysis = new frmWorkGroupAnalysis();
            return locFrmWorkGroupAnalysis.ShowDialog();
        }
    }
}